src/datarobotx/idp/common/credentials_hooks.py

```diff
@@ -25,27 +26,46 @@
 class CredentialsHooks:
     """Include credentials in the catalog so they can be used by nodes."""
 
+    @property
+    def _logger(self):
+        return logging.getLogger(__name__)
+
     @hook_impl
     def after_catalog_created(
         self,
         catalog: DataCatalog,
-        conf_catalog: Dict[str, Any],
         conf_creds: Dict[str, Any],
-        save_version: str,
-        load_versions: Dict[str, str],
     ) -> None:
         """Validate credentials and add them to the catalog for node consumption."""
-        self.validate_credentials(conf_creds)
+        self._validate_or_replace_credentials(conf_creds)
 
-        creds = get_feed_dict(conf_creds, copy_dict_as=None, key_prefix="params:credentials.")
+        creds = get_feed_dict(
+            conf_creds, copy_dict_as=None, key_prefix="params:credentials."
+        )
         catalog.add_feed_dict(creds)
 
-    @staticmethod
-    def validate_credentials(conf_creds: Dict[str, Any]) -> None:
+    def _validate_or_replace_credentials(self, conf_creds: Dict[str, Any]) -> None:
         """Confirm credentials appear to be valid and functional."""
         import datarobot as dr
 
         dr_creds = conf_creds["datarobot"]
+        try:
+            dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
+        except ValueError:
```

Comment:
does this catch all the cases correctly? e.g. key not in dr_creds, key is in dr creds but is not valid for auth, etc?